ffi/athcon/bindings/rust/athcon-vm/src/lib.rs

```diff
@@ -414,6 +435,19 @@ impl TryFrom<&ffi::athcon_message> for ExecutionMessage {
       } else {
         Some(from_buf_raw::<u8>(message.input_data, message.input_size))
       },
+      method: if message.method_name.is_null() {
+        if message.method_name_size != 0 {
+          return Err("msg.method_data is null but msg.method_size is not 0".to_string());
+        }
+        None
+      } else if message.method_name_size == 0 {
+        None
+      } else {
+        Some(from_buf_raw::<u8>(
+          message.method_name,
+          message.method_name_size,
+        ))
```

Comment:
NIT: This and all the uses of `from_buf_raw` could be rewritten without a helper:

```suggestion
        Some(
          unsafe { slice::from_raw_parts(message.method_name, message.method_name_size) }.to_vec(),
        )
```